lib/react/rails/railtie.rb

```diff
@@ -65,7 +65,7 @@ class Railtie < ::Rails::Railtie
       config.after_initialize do |app|
         # Server Rendering
         # Concat component_filenames together for server rendering
-        app.config.react.components_js = lambda {
+        app.config.react.components_js ||= lambda {
```

Comment:
For posterity, there is no difference between an `||=` or an `if` as far as thread safety. If you want to insure a set of instructions is run uninterrupted you have to use a mutex of some kind. Period.  Also, when Rails is initializing, you dont need to worry about threading issues. Those only happen when multiple requests come in to (certain) servers (Puma, etc) which spawn threads to handle each request. By that time the app has already initialized.